### PR TITLE
optimise is_true

### PR DESCRIPTION
**Pull request by DavidBuchanan314**
O(n) -> O(1)  :tada: 

(where n is bits per val)

### CODE REVIEW

**Comment by denballakh** [comment]
lgtm

File: test_simd.py [hunk 1]
```diff
@@ -122,11 +122,11 @@ def test_bit_shift() -> None:
 
 def test_to_bool() -> None:
     assert (
-        A(0x_05_00_03_02_01, *s).is_true() ==
+        A(0x_0f_00_03_02_01, *s).is_true() ==
         A(0x_01_00_01_01_01, *s)
     )
     assert (
-        A(0x_05_00_03_02_01, *s).is_false() ==
+        A(0x_0f_00_03_02_01, *s).is_false() ==
```
Comment (DavidBuchanan314) [inline test_simd.py:129]: btw I tweaked the test case (0x5 -> 0xf) to give better coverage of the all-ones boundary condition
